klippa/tests/integration_test.rs

```diff
@@ -0,0 +1,538 @@
+
+static TEST_DATA_DIR: &str = "./test-data";
+static GEN_EXPECTED_OUTPUTS_VAR: &str = "GEN_EXPECTED_OUTPUTS";
+
+#[derive(Default)]
+struct SubsetTestCase {
+    /// name of directory that stores expected files
+    expected_dir: String,
+
+    /// Path to font files used for testing
+    fonts: Vec<String>,
+
+    /// command line args for subsetter
+    profiles: Vec<String>,
+
+    ///subset codepoints to retain
+    subsets: Vec<String>,
+
+    ///command line args for instancer
+    //TODO: add support for instancing
+    //instances: Vec<String>,
+
+    ///compare against fonttools or not
+    fonttool_options: bool,
+
+    ///IUP optimize or not
+    iup_optimize: Vec<bool>,
+}
+
+#[derive(Default)]
+struct TestCaseParser {
+    case: SubsetTestCase,
+}
+
+/// A helper to iterate over non-empty lines of input
+struct LinesIter<'a> {
+    iter: Peekable<std::str::Lines<'a>>,
+}
+
+impl<'a> LinesIter<'a> {
+    fn new(s: &'a str) -> Self {
+        let mut this = Self {
+            iter: s.lines().peekable(),
+        };
+        this.skip_empty_lines();
+        this
+    }
+
+    fn next(&mut self) -> Option<&'a str> {
+        let next = self.iter.next();
+        self.skip_empty_lines();
+        next
+    }
+
+    fn skip_empty_lines(&mut self) {
+        while let Some(next) = self.iter.peek().copied() {
+            let next = next.trim();
+            if !(next.starts_with('#') || next.is_empty()) {
+                break;
+            }
+            self.iter.next();
+        }
+    }
+
+    fn is_end(&mut self) -> bool {
+        matches!(
+            self.iter.peek().copied(),
+            None | Some(
+                "FONTS:" | "PROFILES:" | "SUBSETS:" | "INSTANCES:" | "OPTIONS:" | "IUP_OPTIONS:",
+            )
+        )
+    }
+}
+
+impl TestCaseParser {
+    fn new() -> Self {
+        TestCaseParser::default()
+    }
+
+    fn parse(mut self, path: &Path) -> SubsetTestCase {
+        let expected_dir = path.file_stem().unwrap().to_str().unwrap();
+        self.case.expected_dir = expected_dir.to_owned();
+
+        let input_text = std::fs::read_to_string(path).unwrap();
+        let mut lines = LinesIter::new(&input_text);
+        while let Some(line) = lines.next() {
+            match line {
+                "FONTS:" => self.parse_fonts(&mut lines),
+                "PROFILES:" => self.parse_profiles(&mut lines),
+                "SUBSETS:" => self.parse_subsets(&mut lines),
+                "INSTANCES:" => self.parse_instances(&mut lines),
+                "OPTIONS:" => self.parse_fonttools_options(&mut lines),
+                "IUP_OPTIONS:" => self.parse_iup_options(&mut lines),
+                other => panic!("unexpected heading '{other}'"),
+            }
+        }
+        self.case
+    }
+
+    fn parse_fonts(&mut self, lines: &mut LinesIter) {
+        while !lines.is_end() {
+            if let Some(next) = lines.next() {
+                self.case.fonts.push(next.to_owned());
```

Comment:
I would probably make all of these `next.trim().to_owned()`, in case there happens to be any unexpected whitespace.